cli/src/semgrep/core_runner.py

```diff
@@ -508,14 +516,15 @@ class CoreRunner:
 
     def __init__(
         self,
-        jobs: int,
+        jobs: Optional[int],
+        deep: bool,
         timeout: int,
         max_memory: int,
         timeout_threshold: int,
         optimizations: str,
         core_opts_str: Optional[str],
     ):
-        self._jobs = jobs
+        self._jobs = jobs if jobs else 1 if deep else __get_cpu_count()
```

Comment:
This is probably fine but would be cleaner to keep get_cpu_count where it was. Separation of concerns. The command line flag parsing at the top level knows what the user wants / the computer default is. Then in this layer we choose to follow or reject that recommendation. (I would also print out something like "number of jobs set to X but setting to 1 because running with --deep"

Reply:
Doing this also has the added benefit of jobs not being undefined all the way down

Reply:
I don't quite understand. If we pass down something which isn't `Optional`, then if we use the `get_cpu_count` up at `scan` time, at the time we produce the `core_runner` we don't know if this recommendation is something the user wants, or if it's the computer default. Both cases behave differently (in the former, we want to take the user's input, and in the latter, we want to override the computer default if `--deep` is present), but they look the same. That's why I have the `Optional`, to disambiguate the case where there is no user input.

My main confusion is that I don't see how we can make the distinction (follow or reject the recommendation) with only a singular integer of data. Or are you saying that `__get_cpu_count` should be moved back, but still used within the context of `core_runner`? My impression was you were saying that it should both be moved back and used at `scan` time.

Reply:
I was mostly suggesting the `--deep` override happen at point closest to semgrep-core invocation 

dont feel too strongly either way on where __get_cpu_count happens :D approving PR

Reply:
oh I think that the confusion is that the desired behavior, as I understand it from Emma, is that we only override if _both_ `deep` is specified and `-j` was NOT passed in. If people really want to run multicore with DeepSemgrep, if they explicitly say to do so, then we should.

Reply:
AH i see.